### ISSUE TITLE
RE.required_fields should be replaced by a validation func

### ISSUE THREAD

**Issue by danielballan** [NSLS-II/bluesky#213]
Current:

```
RE = RunEngine(md={})
```

Prposed:

```
RE = RunEngine(md={}, my_validation_func=None)
```

**Comment by danielballan** [comment]
It's arguable that making owner, group, and beamline_id required in the document schema was a mistake, and that these should not be enforced by mongo but rather by beamline-specific validation functions as above. I especially don't like that the word "beamline" got into this otherwise general experimental data collection package.

**Comment by ericdill** [comment]
:+1: on making it more generic

On Sun, Oct 4, 2015 at 12:10 PM, Dan Allan <notifications@github.com> wrote:

> It's arguable that making owner, group, and beamline_id required in the
> document schema was a mistake, and that these should not be enforced by
> mongo but rather by beamline-specific validation functions as above. I
> especially don't like that the word "beamline" got into this otherwise
> general experimental data collection package.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NSLS-II/bluesky/issues/213#issuecomment-145363278>.
>



-- 
Dr. Eric Dill
Assistant Computational Scientist
National Synchrotron Light Source 2

**Comment by danielballan** [comment]
Closed by #219
